build_tools/benchmarks/export_benchmark_config.py

```diff
@@ -55,52 +56,24 @@
 }
 
 
-def parse_arguments():
-  """Parses command-line options."""
-
-  def parse_and_strip_list_argument(arg) -> List[str]:
-    return [part.strip() for part in arg.split(",")]
-
-  def parse_benchmark_presets(arg) -> List[PresetMatcher]:
-    matchers = []
-    for preset in parse_and_strip_list_argument(arg):
-      matcher = BENCHMARK_PRESET_MATCHERS.get(preset)
-      if matcher is None:
-        raise argparse.ArgumentTypeError(
-            f"Unrecognized benchmark preset: '{preset}'.")
-      matchers.append(matcher)
-    return matchers
-
-  parser = argparse.ArgumentParser(
-      description="Exports JSON config for benchmarking. Filters can be "
-      "specified jointly to select a subset of benchmarks.")
-  parser.add_argument(
-      "--target_device_names",
-      type=parse_and_strip_list_argument,
-      help=("Target device names, separated by comma, not specified means "
-            "including all devices."))
-  parser.add_argument(
-      "--benchmark_presets",
-      type=parse_benchmark_presets,
-      help=("Presets that select a bundle of benchmarks, separated by comma, "
-            "multiple presets will be union. Available options: "
-            f"{','.join(BENCHMARK_PRESET_MATCHERS.keys())}"))
-  parser.add_argument("--output",
-                      type=pathlib.Path,
-                      help="Path to write the JSON output.")
-
-  return parser.parse_args()
-
-
-def main(args: argparse.Namespace):
-  _, all_run_configs = benchmark_collections.generate_benchmarks()
-
-  target_device_names = (set(args.target_device_names)
-                         if args.target_device_names is not None else None)
-  preset_matchers = args.benchmark_presets
-
+def filter_and_group_benchmarks(
```

Comment:
Git diffs the change in an interesting way... But briefly `filter_and_group_benchmarks` and `_benchmark_handler` break out from the original `main`.